### PR TITLE
chore: wip refactor fetch tree

### PR DESCRIPTION
**Pull request by jensneuse**

### CODE REVIEW

File: v2/pkg/engine/postprocess/deduplicate_single_fetches.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+		for j := i + 1; j < len(root.ChildNodes); j++ {
+			if root.ChildNodes[i].Item.Equals(root.ChildNodes[j].Item) {
+				root.ChildNodes = append(root.ChildNodes[:j], root.ChildNodes[j+1:]...)
+				j--
+			}
+		}
+	}
+}
+
+func (d *deduplicateSingleFetches) samePath(a, b *resolve.FetchItem) bool {
```
Comment (devsergiy) [inline v2/pkg/engine/postprocess/deduplicate_single_fetches.go]: unused?

File: v2/pkg/engine/postprocess/order_sequence_by_dependencies.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+	"github.com/wundergraph/graphql-go-tools/v2/pkg/engine/resolve"
+)
+
+type orderSquenceByDependencies struct {
+}
+
+func (o *orderSquenceByDependencies) ProcessFetchTree(root *resolve.FetchTreeNode) {
+	slices.SortFunc(root.ChildNodes, func(_a, _b *resolve.FetchTreeNode) int {
+		a, b := o.nodeFetchID(_a), o.nodeFetchID(_b)
+		aDeps, bDeps := o.nodeDependsOn(_a, root), o.nodeDependsOn(_b, root)
```
Comment (devsergiy) [inline v2/pkg/engine/postprocess/order_sequence_by_dependencies.go:23]: could you add written comments on how ordering works?

File: v2/pkg/engine/resolve/trace.go [hunk 3]
```diff
@@ -195,23 +195,23 @@ func parseFetch(fetch Fetch, options *getTraceOptions) *TraceFetch {
...
 
-	case *ParallelFetch:
-		traceFetch.Type = TraceFetchTypeParallel
-		if f.Trace != nil {
-			traceFetch.Path = f.Trace.Path
-		}
-		for _, subFetch := range f.Fetches {
-			traceFetch.Fetches = append(traceFetch.Fetches, parseFetch(subFetch, options))
-		}
+		/*	case *ParallelFetch:
```
Comment (devsergiy) [inline v2/pkg/engine/resolve/trace.go]: should it be removed?

**Review by devsergiy — CHANGES_REQUESTED** [review]

File: v2/pkg/engine/postprocess/deduplicate_single_fetches.go (hunk 1 above)
Comment (jensneuse) [reply v2/pkg/engine/postprocess/deduplicate_single_fetches.go]: yes

File: v2/pkg/engine/resolve/trace.go (hunk 3 above)
Comment (jensneuse) [reply v2/pkg/engine/resolve/trace.go]: yes

File: v2/pkg/engine/postprocess/order_sequence_by_dependencies.go (hunk 2 above)
Comment (jensneuse) [reply v2/pkg/engine/postprocess/order_sequence_by_dependencies.go:23]: yes

**Review by devsergiy — APPROVED** [review]

**Pull request merged**